### PR TITLE
DOC: Add instructions for setting up development environment on Ubuntu 23.04

### PR DESCRIPTION
**Pull request by mauriliogenovese**
add instruction for lunar lobster

### CODE REVIEW

**Comment by mauriliogenovese** [comment]
instruction for ubuntu lunar lobster

File: Docs/developer_guide/build_instructions/linux.md [hunk 1]
```diff
@@ -76,6 +76,41 @@ sudo apt update && sudo apt install git subversion build-essential cmake cmake-c
...
+tar xf cmake-3.25.3.tar.gz
+
+cd cmake-3.25.3
+./configure
+make
+# Then to install it to your system finally do 
+sudo make install 
+
+# Do this to refresh your console 
+hash -r
```
Comment (pieper) [inline Docs/developer_guide/build_instructions/linux.md:99]: Are these really needed?  It's not even clear to me what they do.

File: Docs/developer_guide/build_instructions/linux.md [hunk 2]
```diff
@@ -76,6 +76,41 @@ sudo apt update && sudo apt install git subversion build-essential cmake cmake-c
   qtbase5-private-dev libqt5x11extras5-dev libxt-dev
 ```
 
+### Ubuntu 23.04 (Lunar Lobster)
+
+:::{warning}
+If you get cmake version error, download build and install cmake 3.25.3:
```
Comment (pieper) [inline Docs/developer_guide/build_instructions/linux.md:82]: I like the idea of getting people the right CMake is good - maybe these instructions below in a separate section about CMake that applies to more than just this ubuntu version, perhaps with notes about how to install the latest CMake on multiple OSes.

File: Docs/developer_guide/build_instructions/linux.md (hunk 1 above)
Comment (pieper) [reply Docs/developer_guide/build_instructions/linux.md:99]: Or maybe it's a typo?

**Review by pieper — CHANGES_REQUESTED** [review]
Thanks for contributing this - just some comments and questions.

File: Docs/developer_guide/build_instructions/linux.md [hunk 3]
```diff
@@ -76,6 +76,41 @@ sudo apt update && sudo apt install git subversion build-essential cmake cmake-c
...
+
+# Then finally check with
+cmake --version 
+```
+:::
+
+Install the development tools and the support libraries:
+
+```console
+sudo apt update && sudo apt install git git-lfs subversion build-essential cmake cmake-curses-gui cmake-qt-gui \
```
Comment (jcfr) [inline Docs/developer_guide/build_instructions/linux.md:109]: Listing `cmake` here and building it above seems to conflict.

Considering that the version of CMake provided by `Ubuntu 23.04` is `3.25.1`[^1] and that the minimum required version by Slicer is `3.16.3`[^2], uninstalling the version provided by the operating system is probably not needed.

[^1]: https://packages.ubuntu.com/lunar/cmake
[^2]: https://github.com/Slicer/Slicer/blob/ba80bddd4a6cba9cfb82a527309cda30ada2ba6a/CMakeLists.txt#L1

**Review by jcfr — CHANGES_REQUESTED** [review]
@mauriliogenovese  Thanks your help for both testing the build process on newer Ubuntu version and improving the build instructions :pray: :100: 

I suggest to revisit the instruction to either build or install the `cmake` package.

If `cmake` needs to be built, we need to explain why and/or update the Slicer build-system to address related issue (if any).

In case we need to update the Slicer build-system, we would be happy to help

File: Docs/developer_guide/build_instructions/linux.md (hunk 1 above)
Comment (mauriliogenovese) [reply Docs/developer_guide/build_instructions/linux.md:99]: Yes it's a typo for bash, maybe we can skip that

File: Docs/developer_guide/build_instructions/linux.md (hunk 3 above)
Comment (mauriliogenovese) [reply Docs/developer_guide/build_instructions/linux.md:109]: When I tried to use the installed cmake I got an error saying that some version were not accepted (including 3.25.1). I have not copied that error but I had to install a newer  release

Comment (jcfr) [reply Docs/developer_guide/build_instructions/linux.md:109]: That is helpful details.

To move forward with this, could you add a section like the following ?

https://github.com/Slicer/Slicer/blob/ba80bddd4a6cba9cfb82a527309cda30ada2ba6a/Docs/developer_guide/build_instructions/linux.md#L48-L51

------

For context, it turns out that CMake `3.25.1` had a regression[^1] that was fixed in version `3.25.2` and above. This is why the default version can't be used.

[^1]: https://gitlab.kitware.com/cmake/cmake/-/issues/24567

File: Docs/developer_guide/build_instructions/linux.md (hunk 1 above)
Comment (jcfr) [inline Docs/developer_guide/build_instructions/linux.md:99]:
```suggestion
```

Removing this section and including a note like we did for `Debian 12 Bookworm (Stable) and Bullseye 11 (OldStable)` should be sufficient.

**Review by jcfr — APPROVED** [review]
I revisited the patch based on the suggestions posted at https://github.com/Slicer/Slicer/pull/7371#discussion_r1389719866

For a preview of the change:

| https://slicer--7371.org.readthedocs.build/en/7371/developer_guide/build_instructions/linux.html#ubuntu-23-04-lunar-lobster |
|--|
| image |

**Review by pieper — APPROVED** [review]
LGTM 👍

**Pull request merged**